assets/src/edit-story/components/library/panes/shared/pillGroup/pillGroup.js

```diff
@@ -84,7 +83,7 @@ const ExpandButton = styled(ArrowDown)`
   align-items: center;
 `;
 
-const Media3pCategories = ({
+const PillGroup = ({
```

Comment:
I have quite a few things I'd like to see cleaned up in this one (didn't review it when it was originally added):

* This component has a few rather complex `useEffect`s - can we just move them to external hooks instead to reduce complexity?
* It has a static id for the pill group, which doesn't work if there's more than one pill group on the DOM at the same time.
* It has some `data-*` attributes too, that I'd prefer to see removed.
* It should have `Boolean(categories.length)` moved to a variable above the return and use a short-circuit conditional rather than ternary: `{hasCategories && <>...</>}` rather than `{hasCategories ? <>...</> : null}`
* I think the `role=tablist` requires a label - either by `aria-labelledby` if we have it in another element or just `aria-label` if we don't.
* The element between `tablist` and `tab` (the `CategoryPillInnerContainer`) should have a `role=presentation` to not confuse screen readers.
* I'm not sure about the purpose of the `renderCategories` function here. It is only used once anyway. It should either be made into a real component or just embedded inline in the JSX.
* It's a bit weird that `title` is sent as a parameter to the `<Pill />` component and not just as children and used as `<Pill ...>{e.displayName}</Pill>`.
* Is `displayName` the best name for the label of a pill? Maybe it should be renamed to just `name`, `title` or `label`.

Reply:
Alright, sounds good, will review all that was already there as well.